### ISSUE TITLE
build fail, no such package '@io_netty_netty_tcnative_boringssl_static//jar'

### ISSUE THREAD

**Issue by fantasticWolff** [iotaledger/compass#142]
```
ERROR: An error occurred during the fetch of repository 'io_netty_netty_tcnative_boringssl_static':
   java.io.IOException: Error downloading [http://central.maven.org/maven2/io/netty/netty-tcnative-boringssl-static/2.0.25.Final/netty-tcnative-boringssl-static-2.0.25.Final.jar] to /home/wolff/.cache/bazel/_bazel_wolff/17f9f494adb64b4a2e8eeff3514ff326/external/io_netty_netty_tcnative_boringssl_static/netty-tcnative-boringssl-static-2.0.25.Final.jar: GET returned 501 HTTPS Required
```
Error downloading, there is 
https://repo1.maven.org/maven2/io/netty/netty-tcnative-boringssl-static/2.0.25.Final/netty-tcnative-boringssl-static-2.0.25.Final.jar
but no
http://central.maven.org/maven2/io/netty/netty-tcnative-boringssl-static/2.0.25.Final/netty-tcnative-boringssl-static-2.0.25.Final.jar

and I don't know how to fix the build with this problem

---
I would like to Set up a private Tangle
[Set up a private Tangle](https://docs.iota.org/docs/compass/0.1/how-to-guides/set-up-a-private-tangle)
and I think this page need to be updated

**Comment by GalRogozinski** [comment]
We will tend to this soon.

In the meanwhile I recommend using [the one command tangle](https://docs.iota.org/docs/utils/0.1/community/one-command-tangle/overview)

**Comment by DaSoty** [comment]
Same error when i install Compass and do the bazel run //docker:layers_calculator

**Comment by loricelli** [comment]
The problem is with the url for that resource. I have the same problem and by following the url you eventually get to this page: [https://support.sonatype.com/hc/en-us/articles/360041287334](url). Basically the http is no more supported and the resources moved to the https version of the site. Let's wait for them to fix it.

**Comment by istabraq-iot** [comment]
I have the same problem:
my error:
```
ERROR: An error occurred during the fetch of repository 'io_netty_netty_tcnative_boringssl_static':
   java.io.IOException: Error downloading [http://central.maven.org/maven2/io/netty/netty-tcnative-boringssl-static/2.0.25.Final/netty-tcnative-boringssl-static-2.0.25.Final.jar] to /home/istabraq/.cache/bazel/_bazel_istabraq/375cc74ae3ea6eac7c6410ee6367e7f6/external/io_netty_netty_tcnative_boringssl_static/netty-tcnative-boringssl-static-2.0.25.Final.jar: Unknown host: central.maven.org
ERROR: /home/istabraq/compass/compass/BUILD:10:1: //compass:layers_calculator depends on @io_netty_netty_tcnative_boringssl_static//jar:jar in repository @io_netty_netty_tcnative_boringssl_static which failed to fetch. no such package '@io_netty_netty_tcnative_boringssl_static//jar': java.io.IOException: Error downloading [http://central.maven.org/maven2/io/netty/netty-tcnative-boringssl-static/2.0.25.Final/netty-tcnative-boringssl-static-2.0.25.Final.jar] to /home/istabraq/.cache/bazel/_bazel_istabraq/375cc74ae3ea6eac7c6410ee6367e7f6/external/io_netty_netty_tcnative_boringssl_static/netty-tcnative-boringssl-static-2.0.25.Final.jar: Unknown host: central.maven.org
ERROR: Analysis of target '//docker:layers_calculator' failed; build aborted: no such package '@io_netty_netty_tcnative_boringssl_static//jar': java.io.IOException: Error downloading [http://central.maven.org/maven2/io/netty/netty-tcnative-boringssl-static/2.0.25.Final/netty-tcnative-boringssl-static-2.0.25.Final.jar] to /home/istabraq/.cache/bazel/_bazel_istabraq/375cc74ae3ea6eac7c6410ee6367e7f6/external/io_netty_netty_tcnative_boringssl_static/netty-tcnative-boringssl-static-2.0.25.Final.jar: Unknown host: central.maven.org
```
how do you solve it ?

**Comment by loricelli** [comment]
> I have the same problem:
> my error:
> 
> ```
> ERROR: An error occurred during the fetch of repository 'io_netty_netty_tcnative_boringssl_static':
>    java.io.IOException: Error downloading [http://central.maven.org/maven2/io/netty/netty-tcnative-boringssl-static/2.0.25.Final/netty-tcnative-boringssl-static-2.0.25.Final.jar] to /home/istabraq/.cache/bazel/_bazel_istabraq/375cc74ae3ea6eac7c6410ee6367e7f6/external/io_netty_netty_tcnative_boringssl_static/netty-tcnative-boringssl-static-2.0.25.Final.jar: Unknown host: central.maven.org
> ERROR: /home/istabraq/compass/compass/BUILD:10:1: //compass:layers_calculator depends on @io_netty_netty_tcnative_boringssl_static//jar:jar in repository @io_netty_netty_tcnative_boringssl_static which failed to fetch. no such package '@io_netty_netty_tcnative_boringssl_static//jar': java.io.IOException: Error downloading [http://central.maven.org/maven2/io/netty/netty-tcnative-boringssl-static/2.0.25.Final/netty-tcnative-boringssl-static-2.0.25.Final.jar] to /home/istabraq/.cache/bazel/_bazel_istabraq/375cc74ae3ea6eac7c6410ee6367e7f6/external/io_netty_netty_tcnative_boringssl_static/netty-tcnative-boringssl-static-2.0.25.Final.jar: Unknown host: central.maven.org
> ERROR: Analysis of target '//docker:layers_calculator' failed; build aborted: no such package '@io_netty_netty_tcnative_boringssl_static//jar': java.io.IOException: Error downloading [http://central.maven.org/maven2/io/netty/netty-tcnative-boringssl-static/2.0.25.Final/netty-tcnative-boringssl-static-2.0.25.Final.jar] to /home/istabraq/.cache/bazel/_bazel_istabraq/375cc74ae3ea6eac7c6410ee6367e7f6/external/io_netty_netty_tcnative_boringssl_static/netty-tcnative-boringssl-static-2.0.25.Final.jar: Unknown host: central.maven.org
> ```
> 
> how do you solve it ?

We have to wait for the developers (the issue has been assigned 3 days ago as you can see from the timeline). I tried but there's no sign of these URLs in the code, so we can't manually change them to the updated ones.

**Comment by GalRogozinski** [comment]
I will look into it on Monday..
The one-command-tangle I mentioned before is not a good work around?

**Comment by istabraq-iot** [comment]
My application use data on Internet so one-command-tangle  does not suitable for me . Thanks for response
@GalRogozinski

**Comment by istabraq-iot** [comment]
> > I have the same problem:
> > my error:
> > ```
> > ERROR: An error occurred during the fetch of repository 'io_netty_netty_tcnative_boringssl_static':
> >    java.io.IOException: Error downloading [http://central.maven.org/maven2/io/netty/netty-tcnative-boringssl-static/2.0.25.Final/netty-tcnative-boringssl-static-2.0.25.Final.jar] to /home/istabraq/.cache/bazel/_bazel_istabraq/375cc74ae3ea6eac7c6410ee6367e7f6/external/io_netty_netty_tcnative_boringssl_static/netty-tcnative-boringssl-static-2.0.25.Final.jar: Unknown host: central.maven.org
> > ERROR: /home/istabraq/compass/compass/BUILD:10:1: //compass:layers_calculator depends on @io_netty_netty_tcnative_boringssl_static//jar:jar in repository @io_netty_netty_tcnative_boringssl_static which failed to fetch. no such package '@io_netty_netty_tcnative_boringssl_static//jar': java.io.IOException: Error downloading [http://central.maven.org/maven2/io/netty/netty-tcnative-boringssl-static/2.0.25.Final/netty-tcnative-boringssl-static-2.0.25.Final.jar] to /home/istabraq/.cache/bazel/_bazel_istabraq/375cc74ae3ea6eac7c6410ee6367e7f6/external/io_netty_netty_tcnative_boringssl_static/netty-tcnative-boringssl-static-2.0.25.Final.jar: Unknown host: central.maven.org
> > ERROR: Analysis of target '//docker:layers_calculator' failed; build aborted: no such package '@io_netty_netty_tcnative_boringssl_static//jar': java.io.IOException: Error downloading [http://central.maven.org/maven2/io/netty/netty-tcnative-boringssl-static/2.0.25.Final/netty-tcnative-boringssl-static-2.0.25.Final.jar] to /home/istabraq/.cache/bazel/_bazel_istabraq/375cc74ae3ea6eac7c6410ee6367e7f6/external/io_netty_netty_tcnative_boringssl_static/netty-tcnative-boringssl-static-2.0.25.Final.jar: Unknown host: central.maven.org
> > ```
> > 
> > 
> > how do you solve it ?
> 
> We have to wait for the developers (the issue has been assigned 3 days ago as you can see from the timeline). I tried but there's no sign of these URLs in the code, so we can't manually change them to the updated ones.

Thanks I will wait too

**Comment by GalRogozinski** [comment]
I tried to have a look today. Due to constraints I couldn't invest too much time on this issue. It seemed to be harder than I anticipated.Please use docker images from here in the meanwhile: https://hub.docker.com/r/iotacafe/compass/tags

**Comment by Exo-dar** [comment]
I had the same issue last week and I've been able to find a workaround.
As @fantasticWolff  stated, the problem was with the url from where Docker was downloading some packages.
Essentially, you have to follow the [Set up a private Tangle](https://docs.iota.org/docs/compass/0.1/how-to-guides/set-up-a-private-tangle) guide. But:
Step 1.2:
Install bazel 0.29 instead of 0.18
wget https://github.com/bazelbuild/bazel/releases/download/0.29.0/bazel-0.29.0-installer-linux-x86_64.sh
Step 2.2:
Before you run "bazel run //docker:layers_calculator" on the compass folder, go to "/home/YOURUSERNAME/.cache/bazel/_bazel_istabraq/375cc74ae3ea6eac7c6410ee6367e7f6/external/io_grpc_grpc_java". Modify the script "repositories.bzl"("nano repositories.bzl"). Look for every line wich starts by "server_urls = ["http://central.maven.org/maven2"]" and change it to "server_urls = ["https://repo1.maven.org/maven2"]". Save the script and go back to the "compass" folder and execute "bazel run //docker:layers_calculator", it should be working now :)

**Comment by istabraq-iot** [comment]
Thanks for answering @Exo-dar 
I have install bazel v 0.28.1, is it ok?
then I have done what you have told me but I have a new problem :( 
```
ERROR: /home/istabraq/compass/compass/BUILD:10:1: //compass:layers_calculator depends on @com_squareup_retrofit2_retrofit//jar:jar in repository @com_squareup_retrofit2_retrofit which failed to fetch. no such package '@com_squareup_retrofit2_retrofit//jar': Failed to fetch Maven dependency: Could not transfer artifact com.squareup.retrofit2:retrofit:jar:2.4.0 from/to com_squareup_retrofit2_retrofit (https://jcenter.bintray.com/): Premature end of Content-Length delimited message body (expected: 91002; received: 35850
ERROR: Analysis of target '//docker:layers_calculator' failed; build aborted: no such package '@com_squareup_retrofit2_retrofit//jar': Failed to fetch Maven dependency: Could not transfer artifact com.squareup.retrofit2:retrofit:jar:2.4.0 from/to com_squareup_retrofit2_retrofit (https://jcenter.bintray.com/): Premature end of Content-Length delimited message body (expected: 91002; received: 35850
```
Could you help me with that please?

**Comment by loricelli** [comment]
Thanks @Exo-dar , I'll try it asap. Just out of curiosity, how did you find out this way to solve it?

**Comment by Exo-dar** [comment]
@istabraq-iot Sorry, but I don't know if it works with bazel v0.28.1. My previous answer wasn't completely accurate, I just changed it. Check it out and tell me if it worked for you :)
@loricelli I was looking at the first lines of my error and that "repositories.bzl" showed up, so I looked into it. Knowing that the repository was wrong, and finding "io_netty_comon" with "server_urls = ["https://central.maven.org/maven2"] I decided to give it a try and (after some work) it worked on my computer.
